Description: Cope with system-config-printer split (now depends only on python-cupsutils)
Origin: vendor: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=525558#21
Bug-Debian: http://bugs.debian.org/525558


Index: kdeadmin-4.2.4/system-config-printer-kde/system-config-printer-kde.py
===================================================================
--- kdeadmin-4.2.4.orig/system-config-printer-kde/system-config-printer-kde.py	2009-06-20 18:55:51.000000000 +0200
+++ kdeadmin-4.2.4/system-config-printer-kde/system-config-printer-kde.py	2009-06-20 18:56:19.000000000 +0200
@@ -32,11 +32,6 @@
 #tempfile
 import thread
 
-#load modules from system-config-printer-common (debug, smburi), change path here if you have it installed elsewhere
-SYSTEM_CONFIG_PRINTER_DIR = "/usr/share/system-config-printer"
-if os.path.exists(SYSTEM_CONFIG_PRINTER_DIR + "/config.py"):
-    sys.path.append(SYSTEM_CONFIG_PRINTER_DIR)
-
 from PyQt4.QtCore import *
 from PyQt4.QtGui import *
 from PyQt4 import uic
@@ -64,10 +59,12 @@
 cups.require ("1.9.27")
 
 # These come from system-config-printer
-import config
 import cupsutils.ppds
-from cupsutils import cupshelpers  #, options
-from smburi import SMBURI
+try:
+    import cupshelpers
+except ImportError:
+    from cupsutils import cupshelpers
+from cupsutils.smburi import SMBURI
 from cupsutils.debug import *
 
 import dbus
@@ -706,8 +703,6 @@
             # otherwise use cups' default one
             custom_testpage = None
             opt = self.ppd.findOption ("PageSize")
-            if opt:
-                custom_testpage = os.path.join(SYSTEM_CONFIG_PRINTER_DIR, 'testpage-%s.ps' % opt.defchoice.lower())
 
             if custom_testpage and os.path.exists(custom_testpage):
                 debugprint ('Printing custom test page ' + custom_testpage)
@@ -1481,9 +1476,8 @@
         """
 
         # Optionally disable downloadable driver support.
-        if not config.DOWNLOADABLE_DRIVER_SUPPORT:
-            self.rbtnNPDownloadableDriverSearch.setEnabled(False)
-            self.downloadableDriverSearchFrame.hide()
+        self.rbtnNPDownloadableDriverSearch.setEnabled(False)
+        self.downloadableDriverSearchFrame.hide()
 
         """
         # Set up OpenPrinting widgets.
